Libraries/react-native/react-native.js

```diff
@@ -10,14 +10,18 @@
  */
 'use strict';
 
+var React = require('React');
+
 // Export React, plus some native additions.
 //
 // The use of Object.create/assign is to work around a Flow bug (#6560135).
 // Once that is fixed, change this back to
 //
-//   var ReactNative = {...require('React'), /* additions */}
+//   var ReactNative = {...React, /* additions */}
 //
-var ReactNative = Object.assign(Object.create(require('React')), {
+var ReactNative = Object.assign(Object.create(React), {
+  React,
```

Comment:
In this instance there's no cycle since react-native is exporting React along with some other native extensions. Another option is remove the Object.assign bit and just have:
```js
// react-native.js
module.exports = {
  React: require('React'),
  // react native components
};
```
and then have the pattern for using it be `var {React, View} = require('react-native')` or alternatively
```js
var ReactNative = require('react-native');
var {React, View} = ReactNative;
```